### PR TITLE
Use logical instead of physical block margins

### PR DESCRIPTION
**Pull request by emonadeo**
I am currently trying to use Capsize with text, that is rotated 90deg (using [`writing-mode: vertical-lr;`](https://developer.mozilla.org/en-US/docs/Web/CSS/writing-mode)). The current implementation always applies `margin-top` and `margin-bottom`, regardless of the actual orientation.

This PR replaces `margin-top` with `margin-block-start` and `margin-bottom` with `margin-block-end`, as these properties respect aforementioned text orientation.

From [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/margin-block#syntax):
> This property corresponds to the [margin-top](https://developer.mozilla.org/en-US/docs/Web/CSS/margin-top) and [margin-bottom](https://developer.mozilla.org/en-US/docs/Web/CSS/margin-bottom), or the [margin-right](https://developer.mozilla.org/en-US/docs/Web/CSS/margin-right) and [margin-left](https://developer.mozilla.org/en-US/docs/Web/CSS/margin-left) properties, depending on the values defined for [writing-mode](https://developer.mozilla.org/en-US/docs/Web/CSS/writing-mode), [direction](https://developer.mozilla.org/en-US/docs/Web/CSS/direction), and [text-orientation](https://developer.mozilla.org/en-US/docs/Web/CSS/text-orientation).

However while this fixes text in vertical writing mode, it does not work with text in vertical writing mode _and_ [upright text orientation](https://developer.mozilla.org/en-US/docs/Web/CSS/text-orientation):

```css
writing-mode: vertical-lr; // or vertical-rl;
text-orientation: upright;
```

This incorrectly _(in the case of capsize)_ applies left and right margins when using start and end.

I personally think breaking this in favor of supporting vertical text is a worth trade-off, but this is of course up for discussion.

I am not sure if this would affect east asian languages that may be written vertically. To my knowledge these are also written horizontally on the web.

### CODE REVIEW

**Comment by michaeltaranto** [comment]
Hey thanks for opening this. We havent been able to use this until very very recently with due to our browser support policy. I wonder if that would be a factor for others too?

Not wanting to add config for no reason, maybe we release this as a breaking change and then if there is a ground swell of people needing the explicit margin direction due to pre Safari 12, then we can add an option then.

How's that sound?

**Comment by emonadeo** [comment]
Sounds good to me. I am down to implement that option if that comes to be the case. Feel free to tag me then, as I probably won't be actively monitoring new issues of this repo.

File: .changeset/nice-flowers-burn.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+Capsize does not work on text, that is rotated 90deg (using [`writing-mode: vertical-lr;`](https://developer.mozilla.org/en-US/docs/Web/CSS/writing-mode)).
+The current implementation always applies `margin-top` and `margin-bottom`, regardless of the actual orientation.
+
+This change replaces `margin-top` with `margin-block-start` and `margin-bottom` with `margin-block-end`, as these properties respect aforementioned text orientation.
+
+From [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/margin-block#syntax):
+> This property corresponds to the [margin-top](https://developer.mozilla.org/en-US/docs/Web/CSS/margin-top) and [margin-bottom](https://developer.mozilla.org/en-US/docs/Web/CSS/margin-bottom), or the [margin-right](https://developer.mozilla.org/en-US/docs/Web/CSS/margin-right) and [margin-left](https://developer.mozilla.org/en-US/docs/Web/CSS/margin-left) properties, depending on the values defined for [writing-mode](https://developer.mozilla.org/en-US/docs/Web/CSS/writing-mode), [direction](https://developer.mozilla.org/en-US/docs/Web/CSS/direction), and [text-orientation](https://developer.mozilla.org/en-US/docs/Web/CSS/text-orientation).
```
Comment (michaeltaranto) [inline .changeset/nice-flowers-burn.md:13]: Might rework this a little as release notes for consumers, rather than PR detail.

```suggestion
Use logical instead of physical block margins

Support different [writing-modes], e.g.`writing-mode: vertical-lr;` by adopting logical block margins in favour of the explicit physical margins.

### BREAKING CHANGE:

#### Browser support

The switch to logical block margins means that browsers from before 2019 will likely not understand the property resulting in no trim. We think it's a valid trade off at this time, get in touch if you disagree.

See [caniuse] for more details on when the different browsers added support.

#### API Change

TL;DR If you are using the object returned from `createStyleObject` as a passthrough without interrogating it's properties, there is no API break!

Given the change to the CSS properties of the pseudo elements (i.e. `marginTop` to `marginBlockStart` and `marginBottom` to `marginBlockEnd`), any consumer of `createStyleObject` that is interrogating the style object may need to update their usage:

```diff
import { createStyleObject } from '@capsizecss/core';

const capsizeStyles = createStyleObject(...);

// Any access to the CSS properties on the return pseudo elements should be updated
- capsizeStyles.['::before'].marginBottom
+ capsizeStyles.['::before'].marginBlockEnd

- capsizeStyles.['::after'].marginTop
+ capsizeStyles.['::after'].marginBlockStart
```

See [margin-block-start] and [margin-block-end] documentation on MDN for more details.

[caniuse]: https://caniuse.com/mdn-css_properties_margin-block-end
[margin-block-start]: https://developer.mozilla.org/en-US/docs/Web/CSS/margin-block-start
[margin-block-end]: https://developer.mozilla.org/en-US/docs/Web/CSS/margin-block-end
[writing-modes]: https://developer.mozilla.org/en-US/docs/Web/CSS/writing-mode
```
